### ISSUE TITLE
A bug

### ISSUE THREAD

**Issue by oleedd** [AdguardTeam/AdguardBrowserExtension#2234]
### AdGuard Extension version

4.1.46

### Browser version

Chrome

### OS version

Windows

### Issue Details

![image](https://user-images.githubusercontent.com/36125949/203365280-480613ec-ee4a-4a40-ad3d-b07a2dfada38.png)
It is from the "Sourses" tab. The first string is from my rule, after this your script.
It may break pages (it is so in my case) while we think that rules are independent.
Solution:
Use `!` instead of `()` or `;`.

**Comment by oleedd** [comment]
Or use separated execution.

**Comment by maximtop** [comment]
Could you describe how to reproduce this issue? 
Seem like the code with error is part of scriptlets

**Comment by oleedd** [comment]
For example:
`picyield.com#%#alert=(a)=>a`
![image](https://user-images.githubusercontent.com/36125949/203806314-70f2bb04-b6da-44f1-8b2f-57c951fc45c2.png)

**Comment by oleedd** [comment]
For example:
`picyield.com#%#alert=a=>a`
![image](https://user-images.githubusercontent.com/36125949/203807667-3f4431ad-d630-4534-8f36-a55f7dfcc0c4.png)

**Comment by stanislav-atr** [comment]
@oleedd hi! Please, specify which filters and user rules you use while doing this, so i can reproduce the error.

**Comment by oleedd** [comment]
Base filter and 1 this rule is enough.

**Comment by maximtop** [comment]
@oleedd adding `;` in the end of you scrip rule should solve the problem
in the future we'd add it ourselves

**Comment by oleedd** [comment]
I know. But it looks like rules are separated and can't interact by syntax while it is not so.
Better use `!` instead of `()`.
And why not separate execution?

**Comment by maximtop** [comment]
>I know. But it looks like rules are separated and can't interact by syntax while it is not so.

I agree with you, we will fix this, for now, you can use the way I've proposed.

> Better use ! instead of ().

I didn't get this point.

> And why not separate execution?

by design

**Comment by oleedd** [comment]
>I didn't get this point.

`!function(){}()` instead of `(function(){})()`.

>by design

What does it mean? Rules are separated (outwardly). I think that separated execution is the most correct (corresponding) solution.

**Comment by maximtop** [comment]
> `!function(){}()` instead of `(function(){})()`.
oh, I see, and why it is better?

> What does it mean? Rules are separated (outwardly). I think that separated execution is the most correct (corresponding) solution.
It means that it has been designed so that all scripts are collected together and executed together. I think that executing scripts separately would increase injection time.

**Comment by maximtop** [comment]
> Because several ` ;` are not logical and standard, and `;` at the beginning of a line too. Checking for `;` at the end is not nice too (additional steps).

using `!` is not standard as well
in my opinion, several `;` is an acceptable solution since it doesn't break the script and its readability

**Comment by oleedd** [comment]
Using of `!` is very common. It is used (at least was, before creating scriplets) in your js rules. I have never seen consecutive `;`. It looks strange. `!` can't break too.

**Comment by oleedd** [comment]
\

**Comment by oleedd** [comment]
Test.
